### PR TITLE
[#133] Design dashboard for current metrics

### PR DESCRIPTION
**Pull request by linzs148**
design dashboard for current metrics and add user document for it

### CODE REVIEW

**Comment by jesperpedersen** [comment]
@linzs148 Please, rebase, and squash your commits
@MariamFahmy98 PTAL

**Comment by linzs148** [comment]
> @linzs148 Please, rebase, and squash your commits

I have removed the merge commit

**Comment by linzs148** [comment]
> @linzs148 Please, rebase, and squash your commits @MariamFahmy98 PTAL

I have removed the merge commit to meke it clean.

**Comment by MariamFahmy98** [comment]
Thanks for the PR. It's good for me.

**Comment by jesperpedersen** [comment]
I think we need an issue for this that can be referenced in the commit message.

And, I think we need some screenshots on the issue or in the discussion thread.

**Comment by jesperpedersen** [comment]
@linzs148 Where are we on https://github.com/pgmoneta/pgmoneta/pull/126#issuecomment-1592827556 ?

**Comment by jesperpedersen** [comment]
@linzs148 ping ?

**Comment by linzs148** [comment]
> @linzs148 ping ?

I have related this pull request with issue [#133] and put screenshots on issue and discussion.

**Comment by jesperpedersen** [comment]
Could you get the feedback sorted this week, please ?

**Comment by linzs148** [comment]
> Could you get the feedback sorted this week, please ?

What does this feedback mean? Comments in this pullrequest or something else?

**Comment by jesperpedersen** [comment]
https://github.com/pgmoneta/pgmoneta/issues/133#issuecomment-1630773387

File: doc/tutorial/10_grafana.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Grafana Dashboard
+
+This tutorial will show you how to encapsulate prometheus standard API and use them to monitor state of pgmoneta with the help of grafana dashboard.
```
Comment (jesperpedersen) [inline doc/tutorial/10_grafana.md:3]: prometheus -> Prometheus
grafana -> Grafana

File: doc/tutorial/10_grafana.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+## Preface
+
+This tutorial assumes that you have an installation of PostgreSQL 10+ and pgmoneta.
+
+See [Install pgmoneta](https://github.com/pgmoneta/pgmoneta/blob/main/doc/tutorial/01_install.md)
+for more detail.
+
+## Prometheus Configuration
+
+First of all, you should install prometheus in your pgmoneta server. For `Ubuntu 22.04`, you can just use `apt-get` command to install it.
```
Comment (jesperpedersen) [inline doc/tutorial/10_grafana.md:14]: prometheus -> Prometheus

Remove "For `Ubuntu 22.04`, you can just use `apt-get` command to install it."

File: doc/tutorial/10_grafana.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+After you successfully install prometheus, you should replace `prometheus.yml ` with content below (For `Ubuntu 22.04`, the whole path is `/etc/prometheus/prometheus.yml`) to configure how to query your pgmoneta metrics.
```
Comment (jesperpedersen) [inline doc/tutorial/10_grafana.md:16]: "installed"
"with the content below"

Remove "(For `Ubuntu 22.04`, the whole path is `/etc/prometheus/prometheus.yml`) to configure how to query your pgmoneta metrics"

File: doc/tutorial/10_grafana.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+After you successfully install prometheus, you should replace `prometheus.yml ` with content below (For `Ubuntu 22.04`, the whole path is `/etc/prometheus/prometheus.yml`) to configure how to query your pgmoneta metrics. 
+```yml
+scrape_configs:
+  - job_name: 'pgmoneta'
+    metrics_path: '/metrics'
+    static_configs:
+      - targets: ['localhost:5001']
+```
+
+Then the prometheus service will query your pgmoneta metrics every 15 seconds and package them as time-series data. You can query your pgmoneta metrics and watch their changes as time passed in prometheus web page (default port is `9090`).
```
Comment (jesperpedersen) [inline doc/tutorial/10_grafana.md:25]: prometheus -> Prometheus

File: doc/tutorial/10_grafana.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+![](https://linzs-upload.oss-cn-shanghai.aliyuncs.com/img/prometheus2.jpg)
```
Comment (jesperpedersen) [inline doc/tutorial/10_grafana.md:26]: Needs to be added to the directory

File: doc/tutorial/10_grafana.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+![](https://linzs-upload.oss-cn-shanghai.aliyuncs.com/img/prometheus3.jpg)
```
Comment (jesperpedersen) [inline doc/tutorial/10_grafana.md:27]: Needs to be added to the directory

File: doc/tutorial/10_grafana.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+## Grafana Dashboard Import
+
+Although prometheus provides capacity of querying and monitoring metrics, we can not customize graph for each metric and place them together.
```
Comment (jesperpedersen) [inline doc/tutorial/10_grafana.md:31]: prometheus -> Prometheus
graph -> graphs

"and place them together" -> "and provide a unified view"

File: doc/tutorial/10_grafana.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+As a result, we use grafana to help us manage all graphs together. First of all, we should install grafana in the computer you need to monitor pgmoneta metrics. You can browse grafana web page with default port `3000`, default user `admin` and default password `admin`. Then you can create prometheus data source of pgmoneta.
```
Comment (jesperpedersen) [inline doc/tutorial/10_grafana.md:33]: grafana -> Grafana

File: doc/tutorial/10_grafana.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+![](https://linzs-upload.oss-cn-shanghai.aliyuncs.com/img/grafana1.jpg)
```
Comment (jesperpedersen) [inline doc/tutorial/10_grafana.md:34]: Needs to be added to the directory

File: doc/tutorial/10_grafana.md [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+Finally you can create dashboard by importing `conmtrib/grafana/dashboard.json` and monitor metrics about pgmoneta.
```
Comment (jesperpedersen) [inline doc/tutorial/10_grafana.md:35]: `conmtrib/` -> `contrib/`

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

**Comment by jesperpedersen** [comment]
Add a link from the `GETTING_STARTED.md` to the tutorial

**Comment by jesperpedersen** [comment]
@linzs148 Please, get this sorted quickly

**Comment by linzs148** [comment]
> @linzs148 Please, get this sorted quickly

@jesperpedersen I have changed the commit and the workflow needs to be approved.

**Comment by jesperpedersen** [comment]
Rebased, and merged.

Thanks for your contribution !

Please, continue discussing what the next steps are in your forum thread.

**Pull request merged**